packages/protocol/contracts/L1/provers/GuardianProver.sol

```diff
@@ -153,17 +181,51 @@ contract GuardianProver is IVerifier, EssentialContract {
         nonReentrant
         returns (bool approved_)
     {
-        bytes32 hash = keccak256(abi.encode(_meta, _tran, _proof.data));
-        approved_ = _approve(_meta.id, hash);
+        bytes32 proofHash = keccak256(abi.encode(_meta, _tran, _proof.data));
+        uint256 _version = version;
+        bytes32 currProofHash = latestProofHash[_version][_meta.id];
+
+        if (currProofHash == 0) {
+            latestProofHash[_version][_meta.id] = proofHash;
+            currProofHash = proofHash;
+        }
 
-        emit GuardianApproval(msg.sender, _meta.id, _tran.blockHash, approved_, _proof.data);
+        bool conflicting = currProofHash != proofHash;
+        bool pauseProving = conflicting && provingAutoPauseEnabled
+            && address(this) == resolve(LibStrings.B_CHAIN_WATCHDOG, true);
 
-        if (approved_) {
-            _deleteApproval(hash);
-            ITaikoL1(resolve(LibStrings.B_TAIKO, false)).proveBlock(
-                _meta.id, abi.encode(_meta, _tran, _proof)
-            );
+        if (conflicting) {
+            latestProofHash[_version][_meta.id] = proofHash;
+            emit ConflictingProofs(_meta.id, msg.sender, currProofHash, proofHash, pauseProving);
         }
+
+        if (pauseProving) {
+            ITaikoL1(resolve(LibStrings.B_TAIKO, false)).pauseProving(true);
+        } else {
+            approved_ = _approve(_meta.id, proofHash);
+            emit GuardianApproval(msg.sender, _meta.id, _tran.blockHash, approved_, _proof.data);
+
+            if (approved_) {
+                delete approvals[_version][proofHash];
+                delete latestProofHash[_version][_meta.id];
+
+                ITaikoL1(resolve(LibStrings.B_TAIKO, false)).proveBlock(
+                    _meta.id, abi.encode(_meta, _tran, _proof)
+                );
+            }
+        }
+    }
+
+    /// @notice Pauses or unpauses the chain proving.
+    /// @param _pause true to pause, false to unpause.
+    function pauseTaikoProving(bool _pause) external whenNotPaused {
+        if (guardianIds[msg.sender] == 0) revert GP_INVALID_GUARDIAN();
+
+        if (address(this) != resolve(LibStrings.B_CHAIN_WATCHDOG, true)) {
+            revert GV_PERMISSION_DENIED();
+        }
+
+        ITaikoL1(resolve(LibStrings.B_TAIKO, false)).pauseProving(_pause);
```

Comment:
Maybe we shouldn't allow unpausing?

Reply:
Good point, actually the chain_watchdog can only unpause. I'll change the code to make sure unpausing is now an option here.

Reply:
Done